superset-frontend/src/explore/components/DisplayQueryButton.jsx

```diff
@@ -156,48 +153,47 @@ export const DisplayQueryButton = props => {
 
   const { slice } = props;
   return (
-    <DropdownButton
-      open={menuVisible}
-      noCaret
+    <Dropdown
+      trigger="click"
       data-test="query-dropdown"
-      title={
-        <span>
-          <i className="fa fa-bars" />
-          &nbsp;
-        </span>
-      }
-      bsSize="sm"
-      pullRight
-      id="query"
-      onToggle={setMenuVisible}
-    >
-      <Menu onClick={handleMenuClick} selectable={false}>
-        {slice && (
-          <Menu.Item key={MENU_KEYS.EDIT_PROPERTIES}>
-            {t('Edit properties')}
+      overlay={
+        <Menu onClick={handleMenuClick} selectable={false}>
+          {slice && (
+            <Menu.Item key={MENU_KEYS.EDIT_PROPERTIES}>
+              {t('Edit properties')}
+            </Menu.Item>
+          )}
+          <Menu.Item>
+            <ModalTrigger
+              triggerNode={
+                <span data-test="view-query-menu-item">{t('View query')}</span>
+              }
+              modalTitle={t('View query')}
+              beforeOpen={() => beforeOpen('query')}
+              modalBody={renderQueryModalBody()}
+              responsive
+            />
           </Menu.Item>
-        )}
-        <Menu.Item>
-          <ModalTrigger
-            triggerNode={
-              <span data-test="view-query-menu-item">{t('View query')}</span>
-            }
-            modalTitle={t('View query')}
-            beforeOpen={() => beforeOpen('query')}
-            modalBody={renderQueryModalBody()}
-            responsive
-          />
-        </Menu.Item>
-        {sqlSupported && (
-          <Menu.Item key={MENU_KEYS.RUN_IN_SQL_LAB}>
-            {t('Run in SQL Lab')}
+          {sqlSupported && (
+            <Menu.Item key={MENU_KEYS.RUN_IN_SQL_LAB}>
+              {t('Run in SQL Lab')}
+            </Menu.Item>
+          )}
+          <Menu.Item key={MENU_KEYS.DOWNLOAD_AS_IMAGE}>
+            {t('Download as image')}
           </Menu.Item>
-        )}
-        <Menu.Item key={MENU_KEYS.DOWNLOAD_AS_IMAGE}>
-          {t('Download as image')}
-        </Menu.Item>
-      </Menu>
-    </DropdownButton>
+        </Menu>
+      }
+    >
+      <div
+        role="button"
+        id="query"
+        tabIndex={0}
+        className="btn btn-default btn-sm"
+      >
```

Comment:
Maybe we can replace this with the shiny new Button component?

Reply:
Agree. Separate PR.